File: web/HIGN/HIGN/Personnel/Interfaces/REST/Transform/CreateExaminerCommandFromResourceAssembler.cs
using HIGN.Personnel.Domain.Model.Commands;
using HIGN.Personnel.Interfaces.REST.Resources;

namespace HIGN.Personnel.Interfaces.REST.Transform;

public class CreateExaminerCommandFromResourceAssembler
{
    public static CreateExaminerCommand ToCommandFromResource(CreateExaminerResource resource)
    {
        return new CreateExaminerCommand(resource.FirstName, resource.LastName, resource.NationalProviderIdentifier);
    }
}

File: openfinalfinal/u202120418/u202120418/src/main/java/org/hign/platform/u202120418/personnel/interfaces/rest/transform/CreateExaminerCommandFromResourceAssembler.java
package org.hign.platform.u202120418.personnel.interfaces.rest.transform;/**
 * @author Fiorella Jarama Peñaloza
 * @version 1.0
 */

import org.hign.platform.u202120418.personnel.domain.model.commands.CreateExaminerCommand;
import org.hign.platform.u202120418.personnel.interfaces.rest.resources.CreateExaminerResource;

/**
 * CreateExaminerCommandFromResourceAssembler - Assembler for the command to create an examiner from a resource
 *
 * @author Fiorella Jarama Peñaloza
 */
public class CreateExaminerCommandFromResourceAssembler {
    /**
     * Transforms a CreateExaminerResource into a CreateExaminerCommand.
     * @param resource the CreateExaminerResource.
     * @return the CreateExaminerCommand.
     */
    public static CreateExaminerCommand toCommandFromResource(CreateExaminerResource resource){
        return new CreateExaminerCommand(
                resource.firstName(),
                resource.lastName(),
                resource.nationalProviderIdentifier());
    }
}
